.github/workflows/ci.yml

```diff
@@ -219,9 +227,59 @@ jobs:
         nameVer=$(echo ${name}/${ver})
         conan upload ${nameVer} -r bg-conan -c
 
+    - name: Upload artifacts
+      if: inputs.store_artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.os }}-executable
+        path: build/
+
+  docker-deploy:
+    # Deploy docker container only when builds succeed
+    # TODO: I have just copied everything Dawood did into here.
+    # Probably we need to generalize it a bit more.
+    needs: [build-and-test]
+    runs-on: ubuntu-22.04
+    if: inputs.docker_deploy && inputs.store_artifacts
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Download bg-search Artifact
+        uses: actions/download-artifact@v3
+        with:
+          # Artifact name
+          name: ubuntu-22.04-executable
+          path: ./Docker/
+
+      - name: See variables
+        run: |
+          echo ${{secrets.BG_SEARCH_DOCKER_REGISTRY_SERVER}} ${{secrets.BG_SEARCH_DOCKER_REGISTRY_NAME}} ${{secrets.BG_SEARCH_DOCKER_REGISTRY_PSWD}}
+
+      - name: See Docker directory
+        run: ls
+
+      - name: Set up Docker Build
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker login
+        uses: docker/login-action@v2
+        with:
+          registry: ${{secrets.BG_SEARCH_DOCKER_REGISTRY_SERVER}}
+          username: ${{secrets.BG_SEARCH_DOCKER_REGISTRY_NAME}}
+          password: ${{secrets.BG_SEARCH_DOCKER_REGISTRY_PSWD}}
```

Comment:
Should make these variable names less specific. Probably just get rid of the `BG_SEARCH` prefix